File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_set<int> dup;

        for(int i = 0; i < nums.size(); i++){

            int curr_num = nums.at(i);
            if (dup.find(curr_num) != dup.end()){
                return true;
            }
            else{
                dup.insert(curr_num);
            }

        }

        return false;

    }
};

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        l = r = 0

        while r < len(nums) - 1:
            
            if not nums[r] == nums[r+1]:
                l += 1
                nums[l] = nums[r+1]
                r += 1
                print(nums)
            else:
                r += 1

        return l+1